File: elixir/gigasecond/lib/gigasecond.ex
defmodule Gigasecond do
  @doc """
  Calculate a date one billion seconds after an input date.
  """
  @spec from({{pos_integer, pos_integer, pos_integer}, {pos_integer, pos_integer, pos_integer}}) ::
          {{pos_integer, pos_integer, pos_integer}, {pos_integer, pos_integer, pos_integer}}
  def from(time) do
    NaiveDateTime.to_erl(NaiveDateTime.add(NaiveDateTime.from_erl!(time), 1000000000))
  end
end

File: cpp/gigasecond/gigasecond.cpp
#include "gigasecond.h"
#include "boost/date_time/posix_time/posix_time.hpp"

using namespace boost::posix_time;
namespace gigasecond {
    ptime advance(ptime input) {
        return input + seconds(1000000000);
    }
}  // namespace gigasecond
